### DISCUSSION TITLE
SEO, search engine optimization for NiceGUI apps

### DISCUSSION THREAD

**Discussion by Shahin-rmz** [zauberzeug/nicegui#451] [Q&A] [answered]
Hello Zauberzug team,
SEO is +60% contents that we set on our website.
Rest is technical parts.
I was wondering how SEO optimized is NiceGUI app and if optimizing the loading speed is what Zauberzeug team also considers about NiceGUI?

Best

**Comment by rodja** [comment, marked as answer]
We have not focussed much on SEO-compatibilty yet. But integration may be challenging. For example [Server-Side Rendering](https://vuejs.org/guide/scaling-up/ssr.html#server-side-rendering-ssr) (SSR) is complicated.

**Comment by Shahin-rmz** [comment]
Thanks for answer.
NiceGUI serves it's purpose perfectly. We can reduce *time to market* of the apps very easily. what I asked was sort of accessory feature.
and it is good that the library supports server side rendering. (Which as non front end pythonist I don't deeply understand it :smile: )

**Comment by rodja** [comment]
I created a separate feature request for SSR: #637

**Comment by Oliviapatrick400** [comment]
If you’re building with NiceGUI and wondering why your app feels invisible on Google it’s probably not your code it’s your SEO game 😅
Search engine optimization for NiceGUI apps is all about making your fast modern UI actually discoverable: clean URLs smart metadata server side rendering where possible and content that search engines can understand.

And honestly this is where teaming up with experts like **[SEO Company Garden Grove](http://seocompanygardengrove.com)** can save you a ton of trial and error so you can focus on shipping features while your traffic actually grows

**Reply by evnchn** [reply]
This is on the edge of self-promotion but Imma let it slide personally because NiceGUI is actually taking a real shot at the SEO game, lead by yours truly. CC @falkoschindler & @rodja because we may have to pay money 💸  (or we can continue to let Claude Code wing it - some SEO is better than no SEO after all?)

Can you read:

- #5813
- #5833 

for additional context? 

Maybe also 

- #637
- #663
- #5494

for additional context.

**Reply by unigrated-solutions** [reply]
@Oliviapatrick400 
I would like to add a counter element to one of the pages on my website. The problem I’m facing is that I want to advertise a 245% performance improvement, but I don’t have the data to substantiate that claim. Please advise.
